source/grid/grid_tools.cc

```diff
@@ -5829,7 +5829,9 @@ namespace GridTools
 
     // Check all points within a given pair of box and cell
     const auto check_all_points_within_box = [&](const auto &leaf) {
-      const auto &box       = leaf.first;
+      const double                relative_tolerance = 1e-12;
+      const BoundingBox<spacedim> box = leaf.first.create_extended(
+        relative_tolerance * leaf.first.side_length(0));
```

Comment:
I think it would be better if we scale each side length with a relative tolerance. Thanks for tackling this! @kronbichler

Reply:
What about creating `BB::create_extended_relative()`?

Reply:
I added @bergbauer's suggestion (via his own commit) to this PR.